operators/pkg/instanceset-landing/handler.go

```diff
@@ -0,0 +1,61 @@
+// Package instanceset_landing contains the main logic and helpers
+// for the instanceset-landing component
+package instanceset_landing
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+// LandingHandler is the main handler of the instanceset-landing: starting from an identifier it (creates, then) redirects to the corresponding instance, if it exists.
+func LandingHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.WriteHeader(405)
+		fmt.Fprintf(w, "Method not allowed")
+		return
+	}
+
+	usernameParam, ok := r.URL.Query()["username"]
+
+	if !ok {
+		w.WriteHeader(400)
+		fmt.Fprint(w, "Invalid request")
+		klog.Errorf("Invalid request: invalid query [%v]", r.URL.Query())
+		return
+	}
+
+	username := strings.Split(usernameParam[0], "@")[0]
+
+	if len(username) < 1 {
+		w.WriteHeader(400)
+		fmt.Fprint(w, "Invalid request")
+		klog.Errorf("Invalid request: invalid username [%v]", usernameParam)
+		return
+	}
+
+	examEndpoint := generateEndpoint(username)
+
+	if Options.DynamicStartup {
+		var err error
```

Comment:
```suggestion
```

Reply:
this is here to create a new examEndpoint below, is there any better way to handle this?

Reply:
Got it. I thought you could just define it below together with the initialization, but I missed you would also shadow at the same time the other `examEndpoint` variable.

Reply:
Found another way